aurora/contracts/src/AuroraErc20FastBridge.sol

```diff
@@ -0,0 +1,457 @@
+        bytes memory args = bytes(
+            string.concat(
+                '{"receiver_id": "',
+                bridgeAddressOnNear,
+                '", "amount": "',
+                Strings.toString(totalTokenAmount),
+                '", "msg": "',
+                initArgsBase64,
+                '"}'
+            )
+        );
+
+        PromiseCreateArgs memory callFtTransfer = _callWithoutTransferWNear(
+            near,
+            transferMessage.transferTokenAddressOnNear,
+            "ft_transfer_call",
+            args,
+            ONE_YOCTO,
+            INIT_TRANSFER_NEAR_GAS
+        );
+        bytes memory callbackArg = abi.encodeWithSelector(
+            this.initTokenTransferCallback.selector,
+            msg.sender,
+            initTransferArgs
+        );
+        PromiseCreateArgs memory callback = near.auroraCall(address(this), callbackArg, NO_DEPOSIT, BASE_NEAR_GAS);
+
+        callFtTransfer.then(callback).transact();
+    }
+
+    function initTokenTransferCallback(
+        address signer,
+        bytes calldata initTransferArgs
+    ) external onlyRole(CALLBACK_ROLE) {
+        uint128 transferredAmount = 0;
+
+        if (AuroraSdk.promiseResult(0).status == PromiseResultStatus.Successful) {
+            transferredAmount = _stringToUint(AuroraSdk.promiseResult(0).output);
+        }
+
+        TransferMessage memory transferMessage = _decodeTransferMessageFromBorsh(initTransferArgs);
+
+        uint128 refundAmount = (transferMessage.transferTokenAmount +
+            transferMessage.feeTokenAmount -
+            transferredAmount);
+        if (refundAmount > 0) {
+            balance[transferMessage.transferTokenAddressOnNear][signer] += refundAmount;
+        }
+
+        string memory initArgsBase64 = Base64.encode(initTransferArgs);
+        emit InitTokenTransfer(
+            signer,
+            initArgsBase64,
+            transferMessage.transferTokenAddressOnNear,
+            transferMessage.transferTokenAmount,
+            transferMessage.feeTokenAmount,
+            transferMessage.recipient,
+            (transferredAmount != 0)
+        );
+    }
+
+    function unlock(uint128 nonce, string calldata proof) external whenNotPaused {
+        bytes memory args = bytes(string.concat('{"nonce": "', Strings.toString(nonce), '", "proof": "', proof, '"}'));
+
+        PromiseCreateArgs memory callUnlock = near.call(
+            bridgeAddressOnNear,
+            "unlock",
+            args,
+            NO_DEPOSIT,
+            UNLOCK_NEAR_GAS
+        );
+        bytes memory callbackArg = abi.encodeWithSelector(this.unlockCallback.selector, msg.sender, nonce);
+        PromiseCreateArgs memory callback = near.auroraCall(address(this), callbackArg, NO_DEPOSIT, BASE_NEAR_GAS);
+
+        callUnlock.then(callback).transact();
+    }
+
+    function unlockCallback(address signer, uint128 nonce) external onlyRole(CALLBACK_ROLE) {
```

Comment:
The `signer` isn't used here

Reply:
fixed: https://github.com/aurora-is-near/fast-bridge-protocol/pull/56/commits/63c6c8e0e13fefccfe44a745910f71fb0fc70213